xtos/include/sof/lib/perf_cnt.h

```diff
@@ -76,9 +76,24 @@ struct perf_cnt_data {
 #define perf_trace_simple(pcd, arg) perf_cnt_trace(arg, pcd)
 
 #if CONFIG_PERFORMANCE_COUNTERS_RUN_AVERAGE
-
 /* perf measurement windows size 2^x */
 #define PERF_CNT_CHECK_WINDOW_SIZE 10
+#define task_perf_avg_info(pcd, task_p, class)					\
+	tr_info(task_p, "perf_cycle task %p, %pU cpu avg %u peak %u",\
+		  class, (class)->uid, \
+		  (uint32_t)((pcd)->cpu_delta_sum),			\
+		  (uint32_t)((pcd)->cpu_delta_peak))
+#define task_perf_cnt_avg(pcd, trace_m, arg, class) do {                             \
+		(pcd)->cpu_delta_sum += (pcd)->cpu_delta_last;          \
+		if (++(pcd)->sample_cnt == 1 << PERF_CNT_CHECK_WINDOW_SIZE) { \
+			(pcd)->cpu_delta_sum >>= PERF_CNT_CHECK_WINDOW_SIZE;      \
```

Comment:
This division could be rounded if no measurable impact to performance. Shift right by one less, add 1, and shift right by one.

Reply:
this is not calculation, it is just used for count 1024 times to print one time log, otherwise, there will be too much logs print.